Sources/RSocketCore/Utility/Closeable.swift

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public protocol Closeable: Disposable {
+
+    /// - Returns a `AnySinglePublisher<Never, Swift.Error>` that terminates when the instance is terminated by any reason. Note, in
+    /// case of error termination, the cause of error will be propagated as an error signal through
+    /// `Subscriber#receive(completion: .failure(cause))`. Otherwise, Subscriber#receive(completion: .finished)} will be called.
+    func onClose() -> AnySinglePublisher<Never, Swift.Error>
```

Comment:
Technically this could be a variable as well, but I'm unsure if it fits the context. It could be implemented as a computed property.
```suggestion
    var onClose: AnySinglePublisher<Never, Swift.Error> { get }
```